### PR TITLE
refactor: better pagination

### PR DESCRIPTION
**Pull request by SteveRedka**
## Checklist

- [x] Tests (if applicable)
- [x] Documentation (if applicable)
- [x] Changelog stub (or use [conventional commit messages](https://www.conventionalcommits.org/))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,6 +185,19 @@ options = {
...
+Initially, `FileList` is a paginated collection. It can be navigated using following methods:
+```ruby
+  @file_list = Uploadcare::Entity::FileList.file_list
+  # Let's assume there are 250 files in cloud. By default, UC loads 100 files. To get next 100 files, do:
+  @next_page = @file_list.next_page
+  # To get previous page:
+  @previous_page = @next_page.previous_page
+  # To attempt to load all objects without pagination:
+  @all_files = @file_list.load.files
+```
```
Comment (dmitry-mukhin) [inline README.md]: User shouldn't bother with pagination. The XXXList object should do it behind the scene when needed.
E.g.:

```python

# there are 250 files total
# and we have a default page size of 100

for f in file_list:  # <-- here
    do_something_with(f)

# on line marked with "here" we'll have total three requests to REST API, when we're
# out of objects on current page:
# [1-100], [101-200], [201-250]
```

Comment (dmitry-mukhin) [reply README.md]: https://github.com/uploadcare/uploadcare-ruby/blob/master/lib/uploadcare/resources/resource_list.rb#L72-L80

Comment (dmitry-mukhin) [reply README.md]: so, this may work as low level API, but users should get a nice way to iterate over object collections

Comment (SteveRedka) [reply README.md]: Added `each`

**Pull request merged**